src/library_fs.js

```diff
@@ -904,7 +904,8 @@ FS.staticInit();` +
       if (!link.node_ops.readlink) {
         throw new FS.ErrnoError({{{ cDefs.EINVAL }}});
       }
-      return PATH_FS.resolve(FS.getPath(link.parent), link.node_ops.readlink(link));
+      var parent = PATH_FS.resolve(FS.getPath(link.parent), link.node_ops.readlink(link)) === path ? link.parent.parent : link.parent;
```

Comment:
Can you construct a test case that shows why this is needed?

Perhaps modify `test/unistd/links.c`.

Also can you explain in more detail why this is needed?  I'm strugglying the understanding why sometimes we resolve relative to one path and sometimes relative to another

Reply:
Thank you for the context, https://github.com/WordPress/wordpress-playground/issues/1283 looks like it has more info

Reply:
@sbc100 The value of the `public/store` symlink when I run `ls -la` in the `public` directory returns an absolute path `/Users/.../test/store` while my test folder name is `test`. 

I used the php method [symlink()](https://www.php.net/manual/en/function.symlink.php) to create the symlink.

I hope this helps.

Reply:
What arguments did you use to create the symlink?   i.e. is it a link to `../store` perhaps?  Or maybe and absolute path?

Reply:
I used this script : 

`test/script.php` 

```
<?php

symlink( __DIR__.'/store', __DIR__.'/public/store' );
```

and ran the script in the `test` folder : `php script.php`

Reply:
I see.. and presumably both of those are absolute paths?  

That means that target of the symbol is absolute (`/full/path/to/public/store`) and not relative (`../public/store`)?   Is that what you want?   Does the problem go away if you use a relative symlink?

Reply:
You're right! It works perfectly with relative paths, but unfortunately, I'm dealing with absolute paths in my current situation. A situation where I am not the one who created the symlinks. Do you think it's still worth pursuing?

Reply:
Yes, this is worth fixing, I'm was just trying to figure out the set of circumstances in which this occurs.

Reply:
I should try different use cases then. My suggestion probably works if the symlink path leads to a file in parent directories. But what if the symlink path leads to children directories within a parent directory. I should investigate.